RandLAPACK/comps/rl_syps.hh

```diff
@@ -22,22 +23,19 @@ class SymmetricPowerSketch {
         virtual int call(
             blas::Uplo uplo,
             int64_t m,
-            const std::vector<T>& A,
+            const T* A,
             int64_t lda,
             int64_t k,
             RandBLAS::RNGState<RNG>& state,
             T*& skop_buff = nullptr,
             T* work_buff = nullptr
         ) = 0;
 
-        virtual RandBLAS::DenseSkOp<T,RNG> call(
-            blas::Uplo uplo,
-            int64_t m,
-            const T* A,
-            int64_t lda,
+        virtual int call(
+            SymmetricLinearOperator<T> &A,
             int64_t k,
-            RandBLAS::RNGState<RNG> state,
-            T* skop_buff = nullptr,
+            RandBLAS::RNGState<RNG>& state,
+            T*& skop_buff = nullptr,
```

Comment:
I'm not sure what a reference to a pointer does, would appreciate some explanation (not necessarily in the code; just so I understand what is happening).

Reply:
@TeachRaccooon made this change. It's so we can allocate memory by an assignment ``skop_buff = new T[...]`` inside the function if ``skop_buff`` is passed in as a null pointer. We end up attaching the memory to a sketching operator that deletes it later on. (How it knows how much to delete, I don't know ... but our address sanitizer (valgrind equivalent) shows no problems with this approach.